exist-core/src/main/java/org/exist/storage/BrokerPool.java

```diff
@@ -975,29 +974,25 @@ public NotificationService getNotificationService() {
      * @return <code>true</code> if transactions can be handled
      */
     public boolean isRecoveryEnabled() {
-        synchronized(readOnly) {
-            return !readOnly && recoveryEnabled;
-        }
+        return !readOnly.get() && recoveryEnabled;
     }
 
     @Override
     public boolean isReadOnly() {
-        synchronized(readOnly) {
-            if(!readOnly) {
-                final long freeSpace = FileUtils.measureFileStore(dataLock.getFile(), FileStore::getUsableSpace);
-                if (freeSpace != -1 && freeSpace < diskSpaceMin) {
-                    LOG.fatal("Partition containing DATA_DIR: {} is running out of disk space [minimum: {} free: {}]. Switching eXist-db into read-only mode to prevent data loss!", dataLock.getFile().toAbsolutePath().toString(), diskSpaceMin, freeSpace);
-                    setReadOnly();
-                }
+        final boolean isReadOnly = readOnly.get();
+        if (!isReadOnly) {
+            final long freeSpace = FileUtils.measureFileStore(dataLock.getFile(), FileStore::getUsableSpace);
+            if (freeSpace != -1 && freeSpace < diskSpaceMin) {
+                LOG.fatal("Partition containing DATA_DIR: {} is running out of disk space [minimum: {} free: {}]. Switching eXist-db into read-only mode to prevent data loss!", dataLock.getFile().toAbsolutePath().toString(), diskSpaceMin, freeSpace);
+                setReadOnly();
```

Comment:
is there a reason why we cannot reset the readonly state?

Reply:
As far as I can tell I have not changed the existing logic here.

If you are asking, if after read-only we can switch back to read-write if disk space has been made available by the admin, then I have no idea without digging deeper into the BufferManager and Journal. However, I think that should be a separate feature request...

Reply:
correct, it was just a question from my side :-)